cunumeric/module.py

```diff
@@ -663,7 +663,7 @@ def arange(
         step = 1
 
     if dtype is None:
-        dtype = np.find_common_type([], [type(start), type(stop), type(step)])
+        dtype = np.result_type(type(start), type(stop), type(step))
```

Comment:
@seberg is this OK or do we need to pass actual scalars?

Reply:
`find_common_type` calls `np.dtype()` first while `np.result_type` tends to convert to arrays when it doens't see an exact scalar type or dtype.  I think that could fail with the `type()` for very strange input.

So it should be a bit safer to pass the values (or use `np.dtype(type(start))` to keep the identical behavior).
Passing values may also be more future-proof, but that goes into too much `arange` weirdness.

Reply:
https://github.com/nv-legate/cunumeric/pull/973/commits/30c4b26498d9acccb3cf3d9d37fa2d4b99345df5